src/value.jl

```diff
@@ -23,7 +23,21 @@ unwrap_catvaluetype(::Type{Any}) = Any # prevent recursion in T>:Missing method
 unwrap_catvaluetype(::Type{T}) where {T <: CategoricalValue} = leveltype(T)
 
 Base.get(x::CategoricalValue) = index(pool(x))[level(x)]
-order(x::CategoricalValue) = order(pool(x))[level(x)]
+
+"""
+    levelindex(x::CategoricalValue)
+
+Get the index of categorical value `x` in the set of possible values
+returned by [`levels(x)`](@ref).
+"""
+levelindex(x::CategoricalValue) = Signed(widen(order(pool(x))[level(x)]))
+
+"""
+    levelindex(x::Missing)
+
+Return `missing`.
+"""
+levelindex(x::Missing) = missing
```

Comment:
Without this you'd have to write `passmissing(levelindex).(x)` on the array which isn't very convenient. Though this kind of goes against the general policy that we only propagate missing values for standard math operators...

Reply:
I think it is OK.